src/patternfly/components/List/examples/List.md

```diff
@@ -77,6 +77,89 @@ cssPrefix: pf-c-list
 {{/list}}
 ```
 
+### With horizontal rules
+```hbs
+{{#> list list--type="ul" list--modifier="pf-m-plain pf-m-bordered"}}
+  <li>Donec blandit a lorem id convallis.</li>
+  <li>Integer in volutpat libero.</li>
+  <li>Donec a diam tellus.</li>
+  <li>Aenean nec tortor orci.</li>
+  <li>Vivamus maximus ultricies pulvinar.</li>
+{{/list}}
+```
+
+### With small icons
+```hbs
+{{#> list list--type="ul" list--modifier="pf-m-plain"}}
+  <li>
+    {{#> list-item-main}}
```

Comment:
Just want to leave a thought that I don't think this is necessary, though I can see it being useful considering the way we've structured other lists like menu/dropown/select/etc, tree view, dual list selector, etc. 

Another way to do it would be to add the class `.pf-m-icon` to the `li`, and that changes the `li` to `display: flex; ...`.

I think I like the way you did it better, since if we want to add a "description" (small grey text under the main item text), we'll just add another element and reuse this structure as opposed to needing a class like `pf-m-description`.

Reply:
Agree w/@mcoker, I think you'd be better served to add `.pf-m-icon` to `li`. I'd use `display: grid` with `column-gap` to easily maintain alignment.

Reply:
@mattnolting what do you think about needing to add a modifier for other things that will use a flex layout here? I think if we're stopping at icons, `pf-m-icon` would be fine. But I can see adding favorites, descriptions, actions, etc (like the other list components) - will we need a modifier for each? If we opt in to the markup @DaoDaoNoCode had previously, you can just place elements (icon, description, etc) in the list and it will lay itself out. Also if we went that route, I would be in favor of placing a margin/padding where we need it instead of using gap so we can just exclude items from the grid if we don't want them as opposed to needing to manage a class or something that removes the `gap`. wdyt?